app/src/main/java/org/schabi/newpipe/settings/SettingsActivity.java

```diff
@@ -91,14 +171,211 @@ public boolean onOptionsItemSelected(final MenuItem item) {
     @Override
     public boolean onPreferenceStartFragment(final PreferenceFragmentCompat caller,
                                              final Preference preference) {
-        final Fragment fragment = Fragment
-                .instantiate(this, preference.getFragment(), preference.getExtras());
+        showSettingsFragment(instantiateFragment(preference.getFragment()));
+        return true;
+    }
+
+    private Fragment instantiateFragment(@NonNull final String className) {
+        return getSupportFragmentManager()
+                .getFragmentFactory()
+                .instantiate(this.getClassLoader(), className);
+    }
+
+    private void showSettingsFragment(final Fragment fragment) {
         getSupportFragmentManager().beginTransaction()
                 .setCustomAnimations(R.animator.custom_fade_in, R.animator.custom_fade_out,
                         R.animator.custom_fade_in, R.animator.custom_fade_out)
-                .replace(R.id.settings_fragment_holder, fragment)
+                .replace(FRAGMENT_HOLDER_ID, fragment)
                 .addToBackStack(null)
                 .commit();
-        return true;
     }
+
+    @Override
+    protected void onDestroy() {
+        setMenuSearchItem(null);
+        searchFragment = null;
+        super.onDestroy();
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Search
+    //////////////////////////////////////////////////////////////////////////*/
+    //region Search
+
+    private void initSearch(
+            final SettingsLayoutBinding settingsLayoutBinding,
+            final boolean restored
+    ) {
+        searchContainer =
+                settingsLayoutBinding.settingsToolbarLayout.toolbar
+                        .findViewById(R.id.toolbar_search_container);
+
+        // Configure input field for search
+        searchEditText = searchContainer.findViewById(R.id.toolbar_search_edit_text);
+        RxTextView.textChanges(searchEditText)
+                // Wait some time after the last input before actually searching
+                .debounce(200, TimeUnit.MILLISECONDS)
+                .subscribe(v -> runOnUiThread(this::onSearchChanged));
+
+        // Configure clear button
+        searchContainer.findViewById(R.id.toolbar_search_clear)
+                .setOnClickListener(ev -> resetSearchText());
+
+        prepareSearchConfig();
+
+        // Build search configuration using SettingsResourceRegistry
+        final PreferenceSearchConfiguration config = new PreferenceSearchConfiguration();
+        SettingsResourceRegistry.getInstance().getAllEntries().stream()
+                .filter(SettingsResourceRegistry.SettingRegistryEntry::isSearchable)
+                .map(SettingsResourceRegistry.SettingRegistryEntry::getPreferencesResId)
+                .forEach(config::index);
```

Comment:
Why do you add items to `config.itemsToIndex` and then only use that list locally in this function? Wouldn't it be better to completely remove the field `itemsToIndex` from `PreferenceSearchConfiguration` and just create a local `ArrayList<SearchIndexItem>`? Or, even better, remove the `forEach` here and replace the `map` just above with `map(entry -> new SearchIndexItem(entry.getPreferencesResId(), config))`. Then you can just append the two other lines of stream manipulation you do below here, i.e. `.map(parser::parse).forEach(searcher::add);`